src/Npgsql/Internal/PgConverter.cs

```diff
@@ -82,6 +83,9 @@ private protected void ThrowIORequired(Size bufferRequirement)
     private protected static bool ThrowInvalidNullValue()
         => throw new ArgumentNullException("value", "Null value given for non-nullable type converter");
 
+    private protected bool ThrowDbNullPredicateOutOfRange()
+        => throw new ArgumentOutOfRangeException(nameof(DbNullPredicateKind), "Unknown case", DbNullPredicateKind.ToString());
```

Comment:
nit: I've shifted to throwing the new UnreachableException for this kind of thing (as it really indicates a bug in the library rather than an invalid argument from the user). We can do a separate pass to change these if we want.

Reply:
I agree as long as the enum value being switched over is not an input. 

I don't think it's a great exception to throw when a value that isn't a valid member of an enum would throw unreachable.

Reply:
Changed!

Reply:
Sure, agree.